src/dbms/dbms_handler.cpp

```diff
@@ -46,7 +47,7 @@ std::string RegisterReplicaErrorToString(query::RegisterReplicaError error) {
 
 // Per storage
 // NOTE Storage will connect to all replicas. Future work might change this
-void RestoreReplication(replication::RoleMainData &mainData, DatabaseAccess db_acc) {
+void RestoreReplication(replication::RoleMainData &mainData, DatabaseAccess db_acc, bool is_coordinator_managed) {
```

Comment:
I don't like this passing booleans into functions. This however is a bit different because we are just propagating it to the constructor of ReplicationStorageClient so I guess it is fine. But my question is why we wouldn't do this part in the ReplicationStorageClient constructor instead of complicating it here?

Reply:
I don't like it too, I organized it differently now, we have singleton object which we can access to check if instance is coordinator managed depending on flags/env